django/contrib/admin/options.py

```diff
@@ -1505,15 +1506,32 @@ def get_changeform_initial_data(self, request):
         Get the initial form data from the request's GET params.
         """
         initial = dict(request.GET.items())
+        formset_data = defaultdict(lambda: defaultdict(defaultdict))
+        formset_query_regex = r'inlinemode_set-(\d+)-(\d+)-(.*)'
```

Comment:
Compile the regular expression here as it'll be used multiple times below. Prefer `[0-9]` to `\d`. Expression should be anchored to avoid risk of partial matches. (Granted, you are using `re.match()` below, but it is always safer to be explicit.)
```python
formset_query_regex = re.compile(r'^inlinemode_set-([0-9]+)-([0-9]+)-(.*)$')
```